### PR TITLE
[IMP] onlogic plm tier validation tests

### PR DESCRIPTION
**Pull request by patrickrwilson**

### CODE REVIEW

File: ol_mrp_plm_tier_validation/tests/test_tier_validation.py [hunk 1]
```diff
@@ -1,48 +1,73 @@
...
             {"state": "progress"}
         )
         cls.normal_stage = cls.env.ref("mrp_plm.ecostage_progress").write(
             {"state": "approved"}
         )
 
+        # Get tier definition model
+        cls.tier_def_obj = cls.env["tier.definition"]
+        # Get ECO model
+        cls.eco_model = cls.env.ref("mrp.model_mrp_eco")
```
Comment (Nikul-OSI) [inline ol_mrp_plm_tier_validation/tests/test_tier_validation.py:28]:
```suggestion
        cls.eco_model = cls.env.ref("mrp_plm.model_mrp_eco")
```

File: ol_mrp_plm_tier_validation/tests/test_tier_validation.py [hunk 2]
```diff
@@ -1,48 +1,73 @@
...
-
-        cls.tier_def_obj = cls.env["tier.definition"]
+        # Set variables from demo data
+        cls.test_user_1 = cls.env.ref("base.partner_admin")
         cls.eco_type = cls.env.ref("mrp_plm.ecotype0")
         cls.product = cls.env.ref(
             "mrp.product_product_computer_desk_bolt_product_template"
         )
+        # Set states on the stages which tier validation uses
         cls.draft_stage = cls.env.ref("mrp_plm.ecostage_new").write(
```
Comment (Nikul-OSI) [inline ol_mrp_plm_tier_validation/tests/test_tier_validation.py:18]: @patrickrwilson Code should be.. please change other accordingly.

cls.draft_stage = cls.env.ref("mrp_plm.ecostage_new")
cls.draft_stage.write(
            {"state": "progress"}
        )

File: ol_mrp_plm_tier_validation/tests/test_tier_validation.py [hunk 3]
```diff
@@ -1,48 +1,73 @@
...
 
     def test_validation_eco(self):
+        """Testing tier validation process on ECO"""
+        # Create an ECO in the 'draft' stage
         eco = self.env["mrp.eco"].create(
             {
                 "name": "ECO for test",
                 "type_id": self.eco_type.id,
                 "type": "product",
                 "product_id": self.product.id,
```
Comment (Nikul-OSI) [inline ol_mrp_plm_tier_validation/tests/test_tier_validation.py:54]:
```suggestion
                "product_tmpl_id": self.product.id,
```

**Comment by patrickrwilson** [comment]
@Nikul-OSI please make the changes, run tests to verify they are working correctly and create a new PR into onl/demo. Thanks for your help with this.

**Comment by Nikul-OSI** [comment]
> @Nikul-OSI please make the changes, run tests to verify they are working correctly and create a new PR into onl/demo. Thanks for your help with this.

https://github.com/ursais/osi-addons/pull/701

**Pull request merged**